torch/csrc/autograd/engine.cpp

```diff
@@ -575,6 +575,7 @@ auto Engine::thread_main(const std::shared_ptr<GraphTask>& graph_task) -> void {
                 local_graph_task->cpu_ready_queue_);
           }
         } catch (std::exception& e) {
+          // See Note [ Persisting PyErr state across autograd engine threads ]
```

Comment:
I meant to put that in the implementation of thread_on_exception in the python_engine: the place where we do persist the error)

Reply:
Oh I was thinking that putting it here would be more visible since when debugging this I didn't even realize that the python engine had overridden this function! Agreed that the implementation is probably the more canonical place though. Going to put in both places.